nsqd/nsqd.go

```diff
@@ -265,6 +269,10 @@ func (n *NSQD) Main() {
 	}
 }
 
+func (n *NSQD) RegisterAddon(addonFn func()) {
+	n.waitGroup.Wrap(func() { addonFn() })
```

Comment:
I might name `RegisterAddon()` slightly differently, like `AddModuleGoroutine()`.

Here's another idea - to avoid needing to add the new public `ExitChan()` method, pass the exitChan to the coroutine function, they'll all need it to integrate with the wait group.

(That'll require going back to creating a new trivial wrapper function like `n.waitGroup.Wrap(func() { modFn(&n.exitChan) })`)